src/storage/chaindb.md

```diff
@@ -0,0 +1,104 @@
+# The ChainDB format
+
+In the Haskell reference implementation, the Storage layer manages a `ChainDB`,
+whose storage components are the `ImmutableDB`, `VolatileDB` and Ledger State
+snapshots:
+
+| Component              | Responsibility                                                                                                                                          |
+|:-----------------------|:--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ImmutableDB            | Store definitive blocks and headers sequentially                                                                                                        |
+| VolatileDB             | Store a bag of non-definitive blocks and headers. In particular it contains the blocks which, when linked sequentially, form the current selected chain |
+| Ledger State snapshots | Periodically store the ledger state at the tip of the ImmutableDB                                                                                       |
+
+Although this implementation represents just one of many possible data storage
+solutions, it is the one used by the reference `cardano-node` implementation and
+has become the de-facto standard for distributing the Cardano chain when the
+node is not involved. Consequently, services like Mithril sign this directory
+structure and its format.
+
+It consists of 3 separate directories in which different data is stored:
+
+```
+db
+├── immutable
+│   ├── 00000.chunk
+│   ├── 00000.primary
+│   ├── 00000.secondary
+│   └── ...
+├── ledger
+│   └── 164021355
+└── volatile
+    ├── blocks-0.dat
+    └── ...
+```
+
+This diagram depicts where the blocks are distributed in such directories:
```

Comment:
Should we add a TODO to (briefly?) discuss chunking?

Reply:
There is a TODO in the `immutable` part

> TODO explain NestedCtxt, Header, Block, the Hard Fork Block, chunks, primary and secondary indices.